security/pkg/k8s/controller/workloadsecret.go

```diff
@@ -255,14 +260,19 @@ func (sc *SecretController) upsertSecret(saName, saNamespace string) {
 	// We retry several times when create secret to mitigate transient network failures.
 	for i := 0; i < secretCreationRetry; i++ {
 		_, err = sc.core.Secrets(saNamespace).Create(secret)
-		if err == nil {
+		if err == nil || errors.IsAlreadyExists(err) {
 			break
 		} else {
 			log.Errorf("Failed to create secret in attempt %v/%v, (error: %s)", i+1, secretCreationRetry, err)
 		}
 		time.Sleep(time.Second)
 	}
 
+	if err != nil && errors.IsAlreadyExists(err) {
+		// Do nothing for the already exist error.
+		return
+	}
+
 	if err != nil {
```

Comment:
replace with `if err != nil && !errors.IsAlreadyExists(err)`

Reply:
Do you think we should throw this info here `log.Infof("Istio secret for service account \"%s\" in namespace \"%s\" has been created", saName, saNamespace)`? if apply your suggestion, this information message shows up.

Reply:
why not log this information?